src/util/simplify_expr.cpp

```diff
@@ -1986,7 +1988,9 @@ simplify_exprt::simplify_byte_extract(const byte_extract_exprt &expr)
     const array_typet &array_type = to_array_type(expr.op().type());
     const auto &element_bit_width =
       pointer_offset_bits(array_type.element_type(), ns);
-    if(element_bit_width.has_value() && *element_bit_width > 0)
+    if(
```

Comment:
again it seems like the optimization would apply less often ?

Reply:
That's just because that very restriction previously was placed earlier in the same function, but now we have some cases that we want to handle later in the same function (where `offset` does not need to be a constant).